File: packages/sdk/output/typescript/src/services/magicLink/MagicLink.ts
import BaseService from '../../BaseService';

import { CreateMagicLinkDto } from './models/CreateMagicLinkDto';

import { serializeQuery } from '../../http/QuerySerializer';

export class MagicLinkService extends BaseService {
  async magicLinkControllerCreateLink(input: CreateMagicLinkDto): Promise<any> {
    const headers: { [key: string]: string } = { 'Content-type': 'application/json' };
    const urlEndpoint = '/magic-link/create';
    const finalUrl = encodeURI(`${this.baseUrl + urlEndpoint}`);
    const response: any = await this.httpClient.post(
      finalUrl,
      input,
      {
        ...headers,
        ...this.getAuthorizationHeader(),
      },
      true,
    );
    const responseModel = response.data;
    return responseModel;
  }

  async magicLinkControllerGetMagicLinks(): Promise<any> {
    const urlEndpoint = '/magic-link';
    const finalUrl = encodeURI(`${this.baseUrl + urlEndpoint}`);
    const response: any = await this.httpClient.get(
      finalUrl,
      {},
      {
        ...this.getAuthorizationHeader(),
      },
      true,
    );
    const responseModel = response.data;
    return responseModel;
  }

  async magicLinkControllerGetMagicLink(id: string): Promise<any> {
    if (id === undefined) {
      throw new Error('The following parameter is required: id, cannot be empty or blank');
    }
    const queryParams: string[] = [];
    if (id) {
      queryParams.push(serializeQuery('form', true, 'id', id));
    }
    const urlEndpoint = '/magic-link/single';
    const finalUrl = encodeURI(`${this.baseUrl + urlEndpoint}?${queryParams.join('&')}`);
    const response: any = await this.httpClient.get(
      finalUrl,
      {},
      {
        ...this.getAuthorizationHeader(),
      },
      true,
    );
    const responseModel = response.data;
    return responseModel;
  }
}

File: packages/sdk/output/python/src/panorasdk/services/magic_link.py
from urllib.parse import quote
from ..net import query_serializer
from .base import BaseService
from ..models.CreateMagicLinkDto import CreateMagicLinkDto as CreateMagicLinkDtoModel


class MagicLink(BaseService):
    def magic_link_controller_create_link(self, request_input: CreateMagicLinkDtoModel):
        url_endpoint = "/magic-link/create"
        headers = {"Content-type": "application/json"}
        self._add_required_headers(headers)

        final_url = self._url_prefix + url_endpoint
        res = self._http.post(final_url, headers, request_input, True)
        return res

    def magic_link_controller_get_magic_links(self):
        url_endpoint = "/magic-link"
        headers = {}
        self._add_required_headers(headers)

        final_url = self._url_prefix + url_endpoint
        res = self._http.get(final_url, headers, True)
        return res

    def magic_link_controller_get_magic_link(self, id: str):
        url_endpoint = "/magic-link/single"
        headers = {}
        query_params = []
        self._add_required_headers(headers)
        if not id:
            raise ValueError("Parameter id is required, cannot be empty or blank.")
        query_params.append(query_serializer.serialize_query("form", False, "id", id))
        final_url = self._url_prefix + url_endpoint + "?" + "&".join(query_params)
        res = self._http.get(final_url, headers, True)
        return res
